addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -297,16 +323,56 @@ export class LinkPlugin extends Plugin {
 
     handleSelectionChange(selectionData) {
         const selection = selectionData.editableSelection;
-        if (!selection.isCollapsed) {
-            this.overlay.close();
-        } else if (!selectionData.documentSelectionIsInEditable) {
+        const props = {
+            onRemove: () => {
+                this.removeLink();
+                this.overlay.close();
+                this.dispatch("ADD_STEP");
+            },
+            onCopy: () => {
+                this.overlay.close();
+            },
+            onClose: () => {
+                this.overlay.close();
+            },
+            getInternalMetaData: this.getInternalMetaData,
+            getExternalMetaData: this.getExternalMetaData,
+        };
+        if (!selectionData.documentSelectionIsInEditable) {
             // note that data-prevent-closing-overlay also used in color picker but link popover
             // and color picker don't open at the same time so it's ok to query like this
             const popoverEl = document.querySelector("[data-prevent-closing-overlay=true]");
             if (popoverEl?.contains(selectionData.documentSelection.anchorNode)) {
                 return;
             }
             this.overlay.close();
+        } else if (!selection.isCollapsed) {
+            const selectedNodes = this.shared.getSelectedNodes();
+            const imageNode = selectedNodes.find((node) => node.tagName === "IMG");
+            if (imageNode && imageNode.parentNode.tagName === "A") {
+                if (this.linkElement !== imageNode.parentElement) {
+                    this.overlay.close();
+                    this.removeCurrentLinkIfEmtpy();
+                }
+                this.linkElement = imageNode.parentElement;
+
+                props.isImage = true;
+                props.linkEl = this.linkElement;
+                props.onApply = (url, _) => {
+                    this.linkElement.href = url;
+                    this.shared.setCursorEnd(this.linkElement);
+                    this.dispatch("ADD_STEP");
+                    this.removeCurrentLinkIfEmtpy();
+                };
```

Comment:
I prefer somethink like (for the mutability)

{ ...props,  isImage:  true;
               linkEl : this.linkElement;
               onApply: (url, _) => {
                    this.linkElement.href = url;
                    this.shared.setCursorEnd(this.linkElement);
                    this.dispatch("ADD_STEP");
                    this.removeCurrentLinkIfEmtpy();
                };}